### PR TITLE
The files needed to generate the graphs.

### PR DESCRIPTION
**Pull request by alex4o**
So I actually have started with using an Ipython notebook because it is easier to work with initially and experiment with different options. This are the parts that will be more stable, so I decided to put them in for review.

### CODE REVIEW

File: extract.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+def get_calls(child):
+    nodes = deque(child.get_children())
+    while len(nodes) > 0:
+        node = nodes.pop()
+        if node.displayname != "" and node.kind == CursorKind.CALL_EXPR:
+            yield node
+        nodes.extend(node.get_children())
+    
+def get_tus(index, path):
```
Comment (ltratt) [inline extract.py:41]: What is `tu`?

**Comment by ltratt** [comment]
Could you give a high-level overview of what the code here is trying to do? Ideally something that we can later put into a commit message.

**Comment by alex4o** [comment]
I am not sure this is sufficient, do you think it should be more in depth explanation?

File: extract.py [hunk 2]
```diff
@@ -1,3 +1,8 @@
+"""
+Extracts the information from C files and
+creates a call graph using igraph.
```
Comment (ltratt) [inline extract.py:3]: Dumb question: what's the output format?

Comment (alex4o) [reply extract.py:3]: That is actually important. Igraph can be used to generate multiple output formats but atm, I don't store the graph on the fs as it is generated fast enough from the c files. So at the moment I save only the svg images.

Comment (ltratt) [reply extract.py:3]: I might be being silly but... what's `Igraph`? :p And is this script producing output that goes straight into Igraph?

**Comment by alex4o** [comment]
Sry this is a thing I should have though of earlier. Should I also include the needed dependencies as a requirements file or just say which packages to install as packages for debian?

File: extract.py (hunk 2 above)
Comment (alex4o) [reply extract.py:3]: This is igraph: https://igraph.org/ and it is not a separate program it is a library for analyzing graphs.

**Comment by ltratt** [comment]
> Should I also include the needed dependencies as a requirements file

I think we need at least the following:

1. A human readable description of what's needed (e.g. in a `README.md`).

and then optionally:

2. A machine readable file to tell pip (or similar) what to install.

**Comment by ltratt** [comment]
@alex4o Is this ready for rereview?

**Comment by alex4o** [comment]
I think so.

**Comment by ltratt** [comment]
Please squash.

**Comment by alex4o** [comment]
Done, there is not build bot for this repo, so you should merge manually.

**Pull request merged**